src/cmd/ksh93/sh/init.c

```diff
@@ -1416,7 +1416,7 @@ Shell_t *sh_init(int argc, char *argv[], Shinit_f userinit) {
 #endif  // SHOPT_TIMEOUT
     // Initialize jobs table.
     job_clear(shp);
-    sh_onoption(shp, SH_MULTILINE);
```

Comment:
This can't possibly be the correct fix. The statement being removed simply makes multiline editing the default behavior. Which is the default in the currently shipping version ksh93u+. Presumably if the user does `set -o multiline` when interacting via the problematic connection the problem will still occur. Is this the actual fix that RH has made to their ksh fork? Unless the person who created this change put a comment in the RH fork or the bugzilla issue that explains why this is correct I think we need to dig deeper.

Reply:
This fix was written more than 4 years ago. But we can still ping @mhlavink who made this change. @mhlavink We would appreciate any comments about it.

Reply:
I have been told that it was part of a memory leak fix which @dgkorn added and the discussion was off list so there is no reference about it in the mailing list.

Reply:
Then I'm inclined to not merge this change. I don't see how changing the default setting of the option would fix a memory leak. There are several mentions of "leak" in *src/cmd/ksh93/RELEASE* but none that would obviously relate to this issue. But I would hope that if @dgkorn and company were aware of a leak that they fixed it in the code that was open sourced. Also, all of the `ksh` commands I have ready access to (which doesn't include any RH flavor) have the `multiline` option enabled by default. Is it really the case that `ksh` on RH distros disable this option by default?

Reply:
This option was turned on by default with last stable release :

```
12-04-26 +Enabled multiline editing by default.  set +o multiline can disable.
```

On RHEL/Fedora it has been turned off by this patch. If you want to investigate this issue further I am fine with not merging it.

Reply:
Given that it's enabled by default in most distros and the fix is both dubious and for an obscure environment that 99.99% of ksh users will never encounter I'd prefer to determine the root cause of the problem and fix that. Anyone using ksh over a "HP ilo4 virtual serial port" can just do `set +o multiline`. Let's open an issue to remind us about the problem and the workaround.